docs/source/getting-started/first-feature.rst

```diff
@@ -0,0 +1,143 @@
+
+Note that there is no explicit check in the choice that the ``sender`` is a friend of the user.
+This is because the ``User`` contract is only ever visible to friends (the observers of the contract).
+
+Now let's see the ``Message`` contract template.
+This is very simple - data and no choices - as well as the ``signatory`` declaration.
+
+.. literalinclude:: code/daml/User.daml
+  :language: daml
+  :start-after: -- MESSAGE_BEGIN
+  :end-before: -- MESSAGE_END
+
+Note that we have two signatories on the ``Message`` contract: both the sender and receiver.
+This enforces the fact that the contract creation (and archival) must be authorized by both parties.
+
+Now we've specified the workflow of sending messages, let's integrate the functionality into our app.
+
+TypeScript Code Generation
+==========================
+
+Remember that we interface with our DAML code from the UI components using the generated TypeScript.
+Since we have changed our DAML code, we also need to rerun the TypeScript code generator.
+Let's do this now by running::
+
+  daml build
+  daml codegen ts .daml/dist/create-daml-app-0.1.0.dar -o daml-ts/src
+
+As the TypeScript code is generated into the separate ``daml-ts`` workspace on which the UI depends, we need to rebuild the workspaces from the root directory using::
+
+  yarn workspaces run build
+
+We should now have the updated TypeScript code with equivalents of the ``Message`` template and ``SendMessage`` choice.
+
+Now let's implement our messaging feature in the UI!
+
+Messaging UI
+============
+
+Our messaging feature has two parts: a form with inputs for selecting friends and composing the message text, and a "feed" of messages that have been sent to you.
+Both parts will be implemented as React components that render on the main screen.
+
+Feed Component
+--------------
+
+The feed component is fairly straight-forward: it queries all ``Message`` contracts and displays their contents as a list.
+Here is the code for the entire component.
+
+.. literalinclude:: code/ui-after/Feed.tsx
+
+The key point here is that for any particular user, the ``Message`` query yields exactly the messages that have been either written by or sent to that user.
+This is due to how we modelled the signatories and observers in the ``Message`` template, and means we do not risk a privacy breach coming from the application code.
+
+Message Edit Component
+----------------------
+
+In addition to the feed component, we need a component for composing messages and sending them using the appropriate choice on the ``User`` contract.
+
+.. literalinclude:: code/ui-after/MessageEdit.tsx
+
+In this component we use React hooks for the message content and receiver.
+You can see these used in the ``submitMessage`` function, called when the "Send" button is clicked.
+The ``isSubmitting`` state is used to ensure that message requests are processed one at a time.
```

Comment:
That is technically not true. This boolean flag is actually completely broken. Let's correct this place once the flag does something useful.